### PR TITLE
Feat: handle Delete messages in join

### PR DESCRIPTION
**Pull request by docmerlin**
This allows `joinNode` to properly handle delete messages.

### CODE REVIEW

File: join.go [hunk 1]
```diff
@@ -91,10 +91,31 @@ func (n *JoinNode) Point(src int, p edge.PointMessage) error {
...
 	g := n.getOrCreateGroup(b.GroupID())
-	g.Barrier(src, b.Time())
+	if err := g.Barrier(src, b.Time()); err != nil {
+		return err
+	}
 	return edge.Forward(n.outs, b)
 }
 
+// Delete deletes the group from the JoinNode, and resets the Low Marks for from the group from that source.
+// if deleteAll is set on the
```
Comment (danxmoran) [inline join.go:101]: Looks like a truncated/unfinished comment

File: integrations/streamer_test.go [hunk 2]
```diff
@@ -4712,6 +4712,61 @@ errorCounts
...
+	|httpOut('TestStream_Delete_Join')
+`
+	er := models.Result{
+		Series: models.Rows{
+			{
+				Name:    "error_view",
+				Tags:    map[string]string{"host": "serverA"},
+				Columns: []string{"time", "error_percent", "errors.sum", "views.sum"},
+				Values: [][]interface{}{
+					{time.Date(1971, 1, 1, 0, 0, 10, 0, time.UTC), 1.0, 18.0, 18.0},
```
Comment (danxmoran) [inline integrations/streamer_test.go:4762]: Making sure I understand the impact: without the changes in the PR, what would be the output of this test script?

**Review by danxmoran — APPROVED** [review]

**Comment by mrabhishekchauhan** [comment]
@docmerlin , how to add the barrier node to joins as this is still not working for me and joins are leaking
also raised an issue : https://community.influxdata.com/t/memory-leak-due-to-join-node/20243

**Comment by m4ce** [comment]
@mrabhishekchauhan - see https://github.com/influxdata/kapacitor/issues/2436. Also, latest 1.6 seems to introduce a regression.

**Pull request merged**